google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/ReplaceableForwardingResultSet.java

```diff
@@ -77,6 +82,21 @@ public boolean next() throws SpannerException {
     return delegate.next();
   }
 
+  @Override
+  public boolean canGetProtobufValue(int columnIndex) {
+    return !closed
+        && delegate instanceof ProtobufResultSet
+        && ((ProtobufResultSet) delegate).canGetProtobufValue(columnIndex);
+  }
+
+  @Override
+  public com.google.protobuf.Value getProtobufValue(int columnIndex) {
+    checkClosed();
+    Preconditions.checkState(
+        delegate instanceof ProtobufResultSet, "The result set does not support protobuf values");
```

Comment:
Cant we call canGetProtobufValue() function to check if it returns true ?

Reply:
No, because `canGetProtobufValue()` is defined in the `ProtobufResultSet` interface, and this check is to verify that the given `ResultSet` is an instance of that (which it is not, which also means that we can't call that method).

Reply:
Oh, wait a second. I think I understand what you mean. Yes, in theory we could also call `this.canGetProtobufValue(..)` (I thought you meant `delegate.canGetProtobufValue(..)`). The reason that we don't do that is that is that `canGetProtobufValue` can also return false for other reasons, for example if the protobuf value is no longer available. That is not the check we want here. Here we want to check if the `ResultSet` at all supports protobuf values (not only: Can I get it __now__, which is what `canGetProtobufValue` returns)